docs: sphinx: Add tiling guidance and performance plots

rleigh-codelibre opened the pull request:
Add tiling documentation to the C++ docs:

- guidance for selecting a tile size
- links to the benchmarking
- plots to demonstrate the effects of varying the tile size

If we wanted, we could generate the plots as part of the build, but it would introduce a dependency upon R and some R packages such as `svglite`, so I've included the generated SVG files alongside the source here.

mtbc commented on docs/sphinx/tiling.rst:
```diff
@@ -0,0 +1,105 @@
+
+When chosing a tile size for TIFF data, consider these factors:
+
+- The tile size must be a multiple of 16
```
these are sentences so maybe end with periods?

mtbc commented on docs/sphinx/tiling.rst:
```diff
@@ -0,0 +1,105 @@
+   <https://github.com/openmicroscopy/ome-files-performance/raw/8f6a8f3b25fd3761967b74366102466a590009f7/analysis/tile-test-write-performance.pdf>`__.
+   It is not necessarily representative of the timings which would be
+   observed in reality, where write caching, buffer copies and other
+   confounding effects which are not modelled here may cause
```
modeled

mtbc commented on docs/sphinx/tiling.rst:
```diff
@@ -0,0 +1,105 @@
+   <https://github.com/openmicroscopy/ome-files-performance/blob/8f6a8f3b25fd3761967b74366102466a590009f7/analysis/strip-test-write-performance.pdf>`__.
+   It is not necessarily representative of the timings which would be
+   observed in reality, where write caching, buffer copies and other
+   confounding effects which are not modelled here may cause
```
modeled

rleigh-codelibre replied:
I've seen the double l plenty in American English.  Is this strictly necessary?

mtbc replied:
*Not* strictly but it's certainly less common, enough that I noticed instead of focusing on the content. E.g., if I try googling `"modelled" site:gov` I get around 217k results, but for "modeled" I get 1,850k results. https://en.wiktionary.org/wiki/modeled says,

> Alternative forms
>     modelled (British)

rleigh-codelibre commented:
I've fixed a few errors in the axis labels, and limited the size of the strip plot to avoid wasting space in the git repo with the svg files (reduced by a quarter).  I rebased it to remove the large files in the history.

dgault commented on docs/sphinx/tiling.rst:
```diff
@@ -0,0 +1,105 @@
+These plots were computed using :download:`an R script
+<plots/tiling.R>`, based upon empirical testing with a `TIFF tiling
+benchmark
+<https://github.com/openmicroscopy/ome-files-performance/commit/8f6a8f3b25fd3761967b74366102466a590009f7>`__.
```
Would it be better to point directly to the doc rather than the commit here?

rleigh-codelibre replied:
I think we would need to create a suitable .md file under `doc/` in the ome-files-performance repo and then link to that file for that specific commit.

dgault commented:
Generally the new tiling doc looks good and reads fine to me. Waiting for https://ci.openmicroscopy.org/view/Failing/job/OME-FILES-CPP-DEV-merge/ to return to normal.

rleigh-codelibre commented on docs/sphinx/tiling.rst:
```diff
@@ -0,0 +1,105 @@
+are shown on the panels as separate lines.
+
+These plots were computed using :download:`an R script
+<plots/tiling.R>`, based upon empirical testing with a `TIFF tiling
```
No.  This script directly generates the files in this directory.  It's the source code for the plots, and is not part of the benchmarking.

rleigh-codelibre commented on docs/sphinx/tiling.rst:
```diff
@@ -0,0 +1,105 @@
+
+   This plot combines the data from the previous plots with the
+   timings derived from `this benchmark result
+   <https://github.com/openmicroscopy/ome-files-performance/raw/8f6a8f3b25fd3761967b74366102466a590009f7/analysis/tile-test-write-performance.pdf>`__.
```
Yes.  Once we have a description of the tests and renamed the tests in the repo, we can tag it and use that tag.

sbesson commented:
Re-reading https://github.com/ome/ome-files-cpp/pull/76#pullrequestreview-47047854, it looks like one of the main reasons for having this excluded was its impact on the CI jobs. With the recent work to use the latest version of Sphinx, this might no longer be a blocker. Assuming no destabilization of the CI jobs, unexcluding this PR and considering it for inclusion in OME Files 0.5.0 should be completely fine from my perspective. /cc @joshmoore

joshmoore commented:
No failing jobs this morning
